protocol/x/clob/keeper/orders.go

```diff
@@ -46,6 +46,83 @@ func (k Keeper) GetOperations(ctx sdk.Context) *types.MsgProposedOperations {
 	return msgProposedOperations
 }
 
+// BatchCancelShortTermOrder removes a specified batch of short term orders from all order-related data
+// structures in the memclob. As well, BatchCancelShortTermOrder adds (or updates) cancels to the desired
+// `goodTilBlock` in the memclob for all specified orders.
+// This message is not atomic. It will optimistically call `CancelShortTermOrder` for every order in the batch.
+// If any of the orders error, the error will be silently logged. This msg will only error if:
+// - Stateful validation fails
+// - Every cancel in the batch fails.
+// This method assumes the provided MsgBatchCancel has already passed ValidateBasic in CheckTx.
+func (k Keeper) BatchCancelShortTermOrder(
+	ctx sdk.Context,
+	msg *types.MsgBatchCancel,
+) error {
```

Comment:
thought about this a little bit more - what do you think about 

1. returning (succeeded []uint32, failed []uint32) here
2. modify `MsgBatchCancelResponse` to return this information to the user (currently it's empty)

This way we don't have to add the logic for `oneCancelSucceeded` and code would be much cleaner, and at the same time, users know exactly which cancels went through. we can also include `[]error` in addition to `failed []uint32`

Reply:
hm i'm down to do the suggestion (1), but returning `[]error` seems a little clunky to me.

Reason being, usually when someone calls a function you expect to check one error to see if it failed or not.
I'm ok with returning just a BatchError, or using the [errors.join](https://pkg.go.dev/errors#Join) method, but it seems to just stack errors together separated by newlines. not sure how i feel about that.

Reply:
Going to punt this. Details in ticket.
https://linear.app/dydx/issue/CT-649/modify-msgbatchcancelresponse-to-return-success-failure-ids

Reply:
I think this function can still return `([]uint32, []uint32)` here, the caller can decide what to do with it (e.g. returning `ErrBatchCancelAllFailed`

Reply:
done